docs/streaming/dev-guide/part2.md

```diff
@@ -0,0 +1,337 @@
+
+**Why synchronous send methods?** Convenience and simplicity. You can call them from anywhere in your async code without `await`:
+
+**Demo Implementation:**
+
+```python
+async def upstream_task() -> None:
+    """Receives messages from WebSocket and sends to LiveRequestQueue."""
+    while True:
+        message = await websocket.receive()
+
+        if "bytes" in message:
+            audio_data = message["bytes"]
+            audio_blob = types.Blob(
+                mime_type="audio/pcm;rate=16000",
+                data=audio_data
+            )
+            live_request_queue.send_realtime(audio_blob)
+
+        elif "text" in message:
+            text_data = message["text"]
+            json_message = json.loads(text_data)
+
+            if json_message.get("type") == "text":
+                content = types.Content(parts=[types.Part(text=json_message["text"])])
+                live_request_queue.send_content(content)
+```
+
+> 📖 **Demo Implementation**: See the complete upstream task in [`app/main.py:129-158`](https://github.com/google/adk-samples/blob/main/python/agents/bidi-demo/app/main.py#L129-L158)
+
+This pattern mixes async I/O operations with sync CPU operations naturally. The send methods return immediately without blocking, allowing your application to stay responsive.
+
+**Important:** This works **only within the same event loop thread**. If you're calling from different threads (e.g., sync FastAPI handlers, background workers), you must use `loop.call_soon_threadsafe()`. See [Cross-Thread Usage](#cross-thread-usage-advanced) below.
+
+!!! note "Best Practice: Create Queue in Async Context"
+
+    Always create `LiveRequestQueue` within an async context (async function or coroutine) to ensure it uses the correct event loop:
+
+    ```python
+    # ✅ Recommended - Create in async context
+    async def main():
+        queue = LiveRequestQueue()  # Uses existing event loop from async context
+        # This is the preferred pattern - ensures queue uses the correct event loop
+        # that will run your streaming operations
+
+    # ❌ Not recommended - Creates event loop automatically
+    queue = LiveRequestQueue()  # Works but ADK auto-creates new loop
+    # This works due to ADK's safety mechanism, but may cause issues with
+    # loop coordination in complex applications or multi-threaded scenarios
+    ```
+
+    **Why this matters:** `LiveRequestQueue` requires an event loop to exist when instantiated. ADK includes a safety mechanism that auto-creates a loop if none exists, but relying on this can cause unexpected behavior in multi-threaded scenarios or with custom event loop configurations.
+
+### Cross-Thread Usage (Advanced)
+
+!!! warning "Not Officially Supported by ADK"
```

Comment:
If this is not officially supported, then you should NOT put it in the official documentation for ADK. Putting a disclaimer on this content does not resolve that issue. 

Remove this content and publish it elsewhere, possibly in the adk-python-community repo. Once this approach is officially sanctioned, we can re-add it here.